postgres_ffi/src/relfile_utils.rs

```diff
@@ -0,0 +1,92 @@
+///
+/// Common utilities for dealing with PostgreSQL relation files.
+///
+use regex::Regex;
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug, Clone)]
+pub struct FilePathError {
+    msg: String,
+}
+
+impl Error for FilePathError {
+    fn description(&self) -> &str {
+        &self.msg
+    }
+}
```

Comment:
This looks like a good place for `thiserror`, maybe something like this:
```rust
#[derive(Debug, Clone, thiserror::Error)]
pub enum FilePathError {
    #[error("invalid filename")]
    InvalidFilename,
}
```

This will automatically derive both `Error` and `Display`.

`thiserror` doesn't have a way to derive `From` while throwing away the source error, so we would still need, e.g.
```rust
impl From<core::num::ParseIntError> for FilePathError {
    fn from(_: core::num::ParseIntError) -> Self {
        FilePathError::InvalidFilename
    }
}
```